rust/lancedb/src/table.rs

```diff
@@ -2023,28 +2024,28 @@ impl TableInternal for NativeTable {
     async fn list_indices(&self) -> Result<Vec<IndexConfig>> {
         let dataset = self.dataset.get().await?;
         let indices = dataset.load_indices().await?;
-        indices.iter().map(|idx| {
-            let mut is_vector = false;
+        futures::stream::iter(indices.as_slice()).then(|idx| async {
```

Comment:
Is it needed? is it common to have a lot of indexes?

Reply:
> Is it needed to use "futures::stream"?

What's the alternative you have in mind? A `while let ...` loop?

> is it common to have a lot of indexes?

For hybrid search you need both an FTS index and a ANN index. And it's common to have a scalar index or two on top of that.

Reply:
I am more thinking of load_indexes would directly get all the index statistics
but may be index_statistics is a call heavier than i expected